diff --git a/tests/runtests.py b/tests/runtests.py
index ccfc327..1ab9ace 100755
--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -20,6 +20,24 @@ except ImportError as e:
         "Django module not found, reference tests/README.rst for instructions."
     ) from e
 else:
+    # Monkeypatch Django template engine with Rusty Templates
+    import django_rusty_templates
+
+    import django.template
+    import django.template.backends.django
+    import django.template.engine
+
+    # Replace Engine class but keep the `get_default` staticmethod
+    # that will return the correct `Engine`
+    django_get_default = django.template.Engine.get_default
+    rust_engine = django_rusty_templates.Engine
+    rust_engine.get_default = django_get_default
+    django.template.Engine = django.template.engine.Engine = rust_engine
+
+    django.template.backends.django.DjangoTemplates = (
+        django_rusty_templates.RustyTemplates
+    )
+
     from django.apps import apps
     from django.conf import settings
     from django.core.exceptions import ImproperlyConfigured
